mlir/include/mlir/Dialect/GPU/IR/CompilationInterfaces.h

```diff
@@ -56,10 +57,11 @@ class TargetOptions {
 
   /// Constructor initializing the toolkit path, the list of files to link to,
   /// extra command line options & the compilation target. The default
-  /// compilation target is `binary`.
+  /// compilation target is `binOrFatbin`.
   TargetOptions(StringRef toolkitPath = {},
                 ArrayRef<std::string> linkFiles = {}, StringRef cmdOptions = {},
-                CompilationTarget compilationTarget = binOrFatbin);
+                CompilationTarget compilationTarget = binOrFatbin,
+                function_ref<SymbolTable *()> symbolTableCallback = {});
```

Comment:
I'd call it `getSymbolTable` (or `getSymbolTableCallback` if you want to include `callback` in the name) to keep a verb here.